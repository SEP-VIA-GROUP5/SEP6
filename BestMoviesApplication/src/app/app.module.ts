import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import {
  NbThemeModule,
  NbLayoutModule,
  NbActionsModule,
  NbSidebarModule,
  NbSidebarService,
  NbMenuModule,
  NbCardModule,
  NbToastrModule,
  NbIconModule,
  NbButtonModule,
  NbSpinnerModule,
  NbListModule,
  NbSearchModule, NbSearchService
} from '@nebular/theme';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { BmLayoutComponent } from '../core/components/bm-layout/bm-layout.component';
import { OverallInformationComponent } from './overall-information/overall-information.component';
import {OverAllInformationState} from "./overall-information/overall-information.state";
import {NgxsModule} from "@ngxs/store";
import {AgGridModule} from "ag-grid-angular";
import { NotFoundComponent } from './not-found/not-found.component';
import {HttpClientModule} from "@angular/common/http";
import {MoviesComponent} from "src/app/information/movies/movies.component";
import {MoviesState} from "src/app/information/movies/movies.state";
import {PeopleComponent} from "src/app/information/people/people.component";
import {PeopleState} from "src/app/information/people/people.state";
import {environment} from "src/environments/environment";
import {ApiService} from "src/core/services/api.service";
import {DirectorsService} from "src/api/directors.service";
import {MoviesService} from "src/api/movies.service";
import {PeopleService} from "src/api/people.service";
import {RatingsService} from "src/api/ratings.service";
import {StarsService} from "src/api/stars.service";
import {MoviesCell} from "src/core/cell-renderers/movies.column.cell";


export const STATES = [
  OverAllInformationState,
  MoviesState,
  PeopleState,
]

export const CELL_RENDERERS = [
  MoviesCell,
]

export const COMPONENTS = [
  AppComponent,

  //components
  BmLayoutComponent,

  //overall information page
  OverallInformationComponent,

  //information page
  MoviesComponent,
  PeopleComponent,

  //404 not found page
  NotFoundComponent,
]

export const PROVIDERS = [
  NbSearchService,

  //Api services
  ApiService,
  DirectorsService,
  MoviesService,
  PeopleService,
  RatingsService,
  StarsService
]

export const NEBULAR_MODULES = [
  NbSidebarModule.forRoot(),
  NbToastrModule.forRoot(),
  NbLayoutModule,
  NbButtonModule,
  NbEvaIconsModule,
  NbActionsModule,
  NbMenuModule,
  NbCardModule,
  NbIconModule,
  NbSpinnerModule,
  NbListModule,
  AgGridModule,
  NbSearchModule,
  NbThemeModule.forRoot({name: 'bm-theme'}),
  NbMenuModule.forRoot(),
]
@NgModule({
  declarations: [
    COMPONENTS,
    ...CELL_RENDERERS
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    NgxsModule.forRoot(STATES, {
      developmentMode: !environment.production,
    }),
    NEBULAR_MODULES,
  ],
  providers: PROVIDERS,
  exports: [
    ...CELL_RENDERERS,
    NEBULAR_MODULES,
  ],
  bootstrap: [AppComponent]
})
export class AppModule { }
