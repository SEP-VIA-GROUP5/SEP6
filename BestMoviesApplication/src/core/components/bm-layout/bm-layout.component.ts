import { Component } from '@angular/core';
import {NbMenuService, NbSidebarService} from '@nebular/theme';
import {
  GENERAL_MENU_ITEMS,
  getIndexOfChildInsideParent,
  getIndexOfParent,
  updateSelectedParentAtIndex,
  updateSelectedChildAtIndexOfParent,
  PARENT_IDS,
} from "src/app/constants";

@Component({
  selector: 'bm-layout',
  template:`
      <nb-layout>
          <nb-layout-header fixed>
            <button
              nbButton
              ghost
              shape="round"
            (click)="toggle()"
            >
              <nb-icon [icon]="isToggled ? 'arrowhead-right-outline' : 'arrowhead-left-outline'"></nb-icon>
            </button>
              <a href="#" (click)="toggle()"><i class="nb-menu"></i></a>
          </nb-layout-header>
          <nb-sidebar>
            <nb-menu [items]="GENERAL_MENU_ITEMS"></nb-menu>
          </nb-sidebar>
          <nb-layout-column class="colored-column-basic">
            <router-outlet></router-outlet>
          </nb-layout-column>
      </nb-layout>
  `
})

export class BmLayoutComponent {

  GENERAL_MENU_ITEMS = GENERAL_MENU_ITEMS;
  selectedItem: any;
  isToggled = false;
  constructor(private sideBarService: NbSidebarService,
              private menu: NbMenuService) {

    let indexOfItem = getIndexOfParent(PARENT_IDS.OVERALL_INFORMATION_ID);
    this.selectedItem = GENERAL_MENU_ITEMS[indexOfItem];
    //event for menu item clicked
    menu.onItemClick().subscribe(item => {
      if(this.selectedItem) {
        if(this.selectedItem.parent) {
          let indexOfItem = getIndexOfChildInsideParent(this.selectedItem.parent.data.id, this.selectedItem.data.id);
          updateSelectedChildAtIndexOfParent(this.selectedItem.parent.data.id, indexOfItem, false);
        }
        else {
          let indexOfItem = getIndexOfParent(this.selectedItem.data.id);
          updateSelectedParentAtIndex(indexOfItem, false);
        }
      }
      item.item.selected = true;
      this.selectedItem = item.item;
    })
  }

  toggle() {
    this.isToggled = !this.isToggled;
    this.sideBarService.toggle(true);
  }
}
