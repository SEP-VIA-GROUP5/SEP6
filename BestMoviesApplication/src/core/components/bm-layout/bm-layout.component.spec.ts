import { BmLayoutComponent } from './bm-layout.component';
import {
  NbActionsModule, NbIconModule,
  NbLayoutModule,
  NbMenuModule,
  NbSidebarModule,
  NbSidebarService,
  NbThemeModule, NbToastrModule, NbToastrService
} from '@nebular/theme';
import { RouterTestingModule } from '@angular/router/testing';
import { BrowserModule } from '@angular/platform-browser';
import { AppRoutingModule } from '../../../app/app-routing.module';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { Spectator } from '@ngneat/spectator';
import { createComponentFactory, SpectatorFactory } from '@ngneat/spectator/jest';
import "jest-extended";
import {NgxsModule} from "@ngxs/store";
import {OverAllInformationState} from "../../../app/overall-information/overall-information.state";
import {PROVIDERS} from "../../../app/app.module";

describe('components > bm-layout > bm-layout.component.spec.ts', () => {
  let component: BmLayoutComponent;
  let spectator: Spectator<BmLayoutComponent>;
  let createComponent: SpectatorFactory<BmLayoutComponent> = createComponentFactory({
    component: BmLayoutComponent,
    imports: [
      RouterTestingModule,
      BrowserModule,
      AppRoutingModule,
      BrowserAnimationsModule,
      NbThemeModule.forRoot({ name: 'bm-theme' }),
      NbSidebarModule.forRoot(),
      NbMenuModule.forRoot(),
      NgxsModule.forRoot([OverAllInformationState]),
      NbToastrModule.forRoot(),
      NbIconModule,
      NbLayoutModule,
      NbEvaIconsModule,
      NbActionsModule,
      NbMenuModule,

    ],
    declarations: [BmLayoutComponent],
    providers: [NbSidebarService,
      ...PROVIDERS],
  });

  beforeEach(() => {
    spectator = createComponent();
  });

  it('should create', () => {
    expect(spectator.component).toBeTruthy();
  });

  it('should match snapshot', async () => {
    spectator.detectChanges();
    await spectator.fixture.whenStable();
    spectator.detectChanges();
    await spectator.fixture.whenRenderingDone();

    // assert
    expect(spectator.fixture).toMatchSnapshot();
  });
});
