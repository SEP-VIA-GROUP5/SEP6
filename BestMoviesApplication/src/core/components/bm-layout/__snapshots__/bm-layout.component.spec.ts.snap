// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > bm-layout > bm-layout.component.spec.ts should match snapshot 1`] = `
<bm-layout
  GENERAL_MENU_ITEMS={[Function Array]}
  isToggled="false"
  menu={[Function NbMenuService]}
  selectedItem={[Function Object]}
  sideBarService={[Function NbSidebarService]}
>
  <nb-layout>
    <div
      class="scrollable-container"
    >
      <div
        class="layout"
      >
        <nb-layout-header
          class="fixed"
          fixed=""
        >
          <nav
            class="fixed"
          >
            <button
              ghost=""
              nbbutton=""
              shape="round"
            >
              <nb-icon>
                <svg
                  class="eva eva-arrowhead-left-outline"
                  fill="currentColor"
                  height="100%"
                  viewBox="0 0 24 24"
                  width="100%"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    data-name="Layer 2"
                  >
                    <g
                      data-name="arrowhead-left"
                    >
                      <rect
                        height="24"
                        opacity="0"
                        transform="rotate(90 12 12)"
                        width="24"
                      />
                      <path
                        d="M11.64 5.23a1 1 0 0 0-1.41.13l-5 6a1 1 0 0 0 0 1.27l4.83 6a1 1 0 0 0 .78.37 1 1 0 0 0 .78-1.63L7.29 12l4.48-5.37a1 1 0 0 0-.13-1.4z"
                      />
                      <path
                        d="M14.29 12l4.48-5.37a1 1 0 0 0-1.54-1.28l-5 6a1 1 0 0 0 0 1.27l4.83 6a1 1 0 0 0 .78.37 1 1 0 0 0 .78-1.63z"
                      />
                    </g>
                  </g>
                </svg>
              </nb-icon>
            </button>
            <a
              href="#"
            >
              <i
                class="nb-menu"
              />
            </a>
          </nav>
        </nb-layout-header>
        <div
          class="layout-container"
        >
          <nb-sidebar
            class="left expanded"
          >
            <div
              class="main-container main-container-fixed"
            >
              <div
                class="scrollable"
              >
                <nb-menu>
                  <ul
                    class="menu-items"
                  >
                    <li
                      class="menu-item"
                      nbmenuitem=""
                    >
                      <a
                        href="http://localhost/overall-information"
                        title="Overall"
                      >
                        <nb-icon
                          class="menu-icon"
                        >
                          <svg
                            class="eva eva-map-outline"
                            fill="currentColor"
                            height="100%"
                            viewBox="0 0 24 24"
                            width="100%"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              data-name="Layer 2"
                            >
                              <g
                                data-name="map"
                              >
                                <rect
                                  height="24"
                                  opacity="0"
                                  transform="rotate(180 12 12)"
                                  width="24"
                                />
                                <path
                                  d="M20.41 5.89l-4-1.8H15.59L12 5.7 8.41 4.09h-.05L8.24 4h-.6l-4 1.8a1 1 0 0 0-.64 1V19a1 1 0 0 0 .46.84A1 1 0 0 0 4 20a1 1 0 0 0 .41-.09L8 18.3l3.59 1.61h.05a.85.85 0 0 0 .72 0h.05L16 18.3l3.59 1.61A1 1 0 0 0 20 20a1 1 0 0 0 .54-.16A1 1 0 0 0 21 19V6.8a1 1 0 0 0-.59-.91zM5 7.44l2-.89v10l-2 .89zm4-.89l2 .89v10l-2-.89zm4 .89l2-.89v10l-2 .89zm6 10l-2-.89v-10l2 .89z"
                                />
                              </g>
                            </g>
                          </svg>
                        </nb-icon>
                        <span
                          class="menu-title"
                        >
                          Overall
                        </span>
                      </a>
                    </li>
                    <li
                      class="menu-item"
                      nbmenuitem=""
                    >
                      <a
                        aria-expanded="false"
                        href="#"
                        title="Information"
                      >
                        <nb-icon
                          class="menu-icon"
                        >
                          <svg
                            class="eva eva-info-outline"
                            fill="currentColor"
                            height="100%"
                            viewBox="0 0 24 24"
                            width="100%"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              data-name="Layer 2"
                            >
                              <g
                                data-name="info"
                              >
                                <rect
                                  height="24"
                                  opacity="0"
                                  transform="rotate(180 12 12)"
                                  width="24"
                                />
                                <path
                                  d="M12 2a10 10 0 1 0 10 10A10 10 0 0 0 12 2zm0 18a8 8 0 1 1 8-8 8 8 0 0 1-8 8z"
                                />
                                <circle
                                  cx="12"
                                  cy="8"
                                  r="1"
                                />
                                <path
                                  d="M12 10a1 1 0 0 0-1 1v5a1 1 0 0 0 2 0v-5a1 1 0 0 0-1-1z"
                                />
                              </g>
                            </g>
                          </svg>
                        </nb-icon>
                        <span
                          class="menu-title"
                        >
                          Information
                        </span>
                        <nb-icon
                          class="expand-state"
                          pack="nebular-essentials"
                        >
                          <svg
                            fill="currentColor"
                            height="100%"
                            viewBox="0 0 24 24"
                            width="100%"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              data-name="Layer 2"
                            >
                              <g
                                data-name="chevron-left"
                              >
                                <rect
                                  height="24"
                                  opacity="0"
                                  transform="rotate(90 12 12)"
                                  width="24"
                                />
                                <path
                                  d="M13.36 17a1 1 0 0 1-.72-.31l-3.86-4a1 1 0 0 1 0-1.4l4-4a1 1 0 1 1 1.42 1.42L10.9 12l3.18 3.3a1 1 0 0 1 0 1.41 1 1 0 0 1-.72.29z"
                                />
                              </g>
                            </g>
                          </svg>
                        </nb-icon>
                      </a>
                      <ul
                        class="menu-items collapsed ng-trigger ng-trigger-toggle"
                        style="height:0px;margin:0px;0:height;1:margin;"
                      >
                        <li
                          class="menu-item"
                          nbmenuitem=""
                          style=""
                        >
                          <a
                            href="http://localhost/information/movies"
                            title="Movies"
                          >
                            <nb-icon
                              class="menu-icon"
                            >
                              <svg
                                class="eva eva-film-outline"
                                fill="currentColor"
                                height="100%"
                                viewBox="0 0 24 24"
                                width="100%"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  data-name="Layer 2"
                                >
                                  <g
                                    data-name="film"
                                  >
                                    <rect
                                      height="24"
                                      opacity="0"
                                      width="24"
                                    />
                                    <path
                                      d="M18.26 3H5.74A2.74 2.74 0 0 0 3 5.74v12.52A2.74 2.74 0 0 0 5.74 21h12.52A2.74 2.74 0 0 0 21 18.26V5.74A2.74 2.74 0 0 0 18.26 3zM7 11H5V9h2zm-2 2h2v2H5zm4-8h6v14H9zm10 6h-2V9h2zm-2 2h2v2h-2zm2-7.26V7h-2V5h1.26a.74.74 0 0 1 .74.74zM5.74 5H7v2H5V5.74A.74.74 0 0 1 5.74 5zM5 18.26V17h2v2H5.74a.74.74 0 0 1-.74-.74zm14 0a.74.74 0 0 1-.74.74H17v-2h2z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </nb-icon>
                            <span
                              class="menu-title"
                            >
                              Movies
                            </span>
                          </a>
                        </li>
                        <li
                          class="menu-item"
                          nbmenuitem=""
                          style=""
                        >
                          <a
                            href="http://localhost/information/people"
                            title="People"
                          >
                            <nb-icon
                              class="menu-icon"
                            >
                              <svg
                                class="eva eva-people-outline"
                                fill="currentColor"
                                height="100%"
                                viewBox="0 0 24 24"
                                width="100%"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <g
                                  data-name="Layer 2"
                                >
                                  <g
                                    data-name="people"
                                  >
                                    <rect
                                      height="24"
                                      opacity="0"
                                      width="24"
                                    />
                                    <path
                                      d="M9 11a4 4 0 1 0-4-4 4 4 0 0 0 4 4zm0-6a2 2 0 1 1-2 2 2 2 0 0 1 2-2z"
                                    />
                                    <path
                                      d="M17 13a3 3 0 1 0-3-3 3 3 0 0 0 3 3zm0-4a1 1 0 1 1-1 1 1 1 0 0 1 1-1z"
                                    />
                                    <path
                                      d="M17 14a5 5 0 0 0-3.06 1.05A7 7 0 0 0 2 20a1 1 0 0 0 2 0 5 5 0 0 1 10 0 1 1 0 0 0 2 0 6.9 6.9 0 0 0-.86-3.35A3 3 0 0 1 20 19a1 1 0 0 0 2 0 5 5 0 0 0-5-5z"
                                    />
                                  </g>
                                </g>
                              </svg>
                            </nb-icon>
                            <span
                              class="menu-title"
                            >
                              People
                            </span>
                          </a>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </nb-menu>
              </div>
            </div>
          </nb-sidebar>
          <div
            class="content"
          >
            <div
              class="columns"
            >
              <nb-layout-column
                class="colored-column-basic"
              >
                <router-outlet />
              </nb-layout-column>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nb-layout>
</bm-layout>
`;
